### DISCUSSION TITLE
What about to use `lastChild` and `previousSibling` as well?

### DISCUSSION THREAD

**Discussion by ghost** [solidjs/solid#644] [Ideas]
I've been thinking about it since I've heard about the Solid's approach that uses `nextSibling` and `firstChild`.

Becase sometimes it's not efficient to go top -> bottom, you wanna go bottom -> top.

The diagram below shows paths that all these four properties create `lastChild` .`firstChild`.`nextSib`.`prevSib`
Let's say we want to get from 0 to 5, using only nextSibling would cost us 5, when with the route 0-5-1 only 2.

![img](https://github.com/pheianox/solid-modal-issue-test/blob/main/Screenshot%20(2).png?raw=true)

In the example below it's better to use `lastChild` property on the parent rather than creating a bunch of `nextSibling` references.

```jsx

<div>
    <h1>Hi mom, jokes on you</h1>
    <h1>Hi mom, jokes on you</h1>
    <h1>Hi mom, jokes on you</h1>
    <h1>Hi mom, jokes on you</h1>
    <h1>Hi mom, jokes on you</h1>
    <h1>Hi mom, jokes on you</h1>
    <h1>Hi mom, jokes on you</h1>
    <h1>Hi mom, jokes on you</h1>
    <button onclick={callPolice}>Call The Police</button>
</div>
```

What about this bad boy?
```jsx

<div>
    <h1>Hi cj, jokes on you</h1>
    <h1>Hi mia, jokes on you</h1>
    <h1>Hi steve, jokes on you</h1>
    <h1>Hi solid, jokes on you</h1>
    <h1>Hi daddy, jokes on you</h1>
    <h1>Hi papa, jokes on you</h1>
    <h1>Hi natasha, jokes on you</h1>
    <h1>Hi spiderman, jokes on you</h1>
    ... +99
    <button onclick={callPolice}>Call The Police</button>
   <h1>Dead mouse</h1>   
   <h1>Don't crash the car</h1>
</div>
```
Well, in this case we can say, yo mama, I don't care, I'm gonna do `div.lastChild.previousSibling`.

And the algorithm is quite simple compare top bottom nodes choose the shortest path and that's it. Of course, we can go deeper to math using Dijkstra's multiple path algorightms with the power of Graphs it's quite simple, but it's really worth it.

**Comment by ryansolid** [comment]
Yes I think this is a place where improvements can be made. The current walks are built quite naively with traversal, but analyzing shape could yield better results.
